fastai2/callback/wandb.py

```diff
@@ -72,31 +83,66 @@ def after_epoch(self):
     def after_fit(self):
         self.run = True
         if self.log_preds: self.remove_cb(self.learn.fetch_preds)
-        wandb.log({}) #To trigger one last synch
+
+    def _log_config(self):
+        "Log configuration parameters"
+        config={}
+
+        # log callbacks
+        try: config={f'{cb}':True for cb in self.cbs}
+        except: print(f'Could not set wandb config callbacks')
+
+        # log input dimensions
+        try:
+            xb = self.dls.train.one_batch()[:self.dls.train.n_inp]
+            config.update({f'input dim {i}':d for i,d in enumerate(list(detuplify(xb).shape))})
+        except: print(f'Could not set wandb config input dimensions')
+        _try_set(config, 'batch size', self.dls.bs)
+        _try_set(config, 'batch per epoch', len(self.dls.train))
+        _try_set(config, 'model', self.model.__class__.__name__)
+        _try_set(config, 'model parameters', total_params(self.model)[0])
+        _try_set(config, 'loss function', f'{self.loss_func}')
+        _try_set(config, 'device', self.dls.device.type)
+        _try_set(config, 'optimizer', self.opt_func.__name__)
+        _try_set(config, 'frozen', bool(self.opt.frozen_idx))
+        _try_set(config, 'frozen idx', self.opt.frozen_idx)
+        _try_set(config, 'dataset.tfms', f'{self.dls.dataset.tfms}')
+        _try_set(config, 'dls.after_item', f'{self.dls.after_item}')
+        _try_set(config, 'dls.before_batch', f'{self.dls.before_batch}')
+        _try_set(config, 'dls.after_batch', f'{self.dls.after_batch}')
+
+        wandb.config.update(config, allow_val_change=True) # in case callback runs multiple times
+
+
+# Cell
+def make_plt(img):
```

Comment:
This function could be private, I think.

Reply:
Ok, if you want I can also put `_make_plt` and `_try_set` as static methods within `WandbCallback` (I don't think they would be needed for any other function).